curator-client/src/main/java/org/apache/curator/ensemble/exhibitor/ExhibitorEnsembleProvider.java

```diff
@@ -301,7 +301,7 @@ else if ( port != thePort )
                     values.putAll(decodeExhibitorList(encoded));
                     done = true;
                 }
-                catch ( Throwable e )
+                catch ( Exception e )
```

Comment:
Probably not. But, I'd view that as a separate issue. It's technically up to the retry policy passed to ExhibitorEnsembleProvider.